src/main/scala/com/krykowski/service/TaskService.scala

```diff
@@ -0,0 +1,15 @@
+package com.krykowski.service
+
+import cats.effect.IO
+import com.krykowski.model
+import com.krykowski.model.{CsvFileLocation, Task, TaskNotFoundError}
+
+trait TaskService {
+  def createTask(csv: CsvFileLocation): IO[Long]
+
+  def getAllTasks(): List[Task]
+
+  def cancelTask(taskId: Long): IO[Either[TaskNotFoundError.type, Unit]]
+
+  def getJson(taskId: Long): IO[Either[model.TaskNotFoundError.type, List[String]]]
```

Comment:
One of critical requirements of the challenge is, that processed files might not fit into the memory and they must be streamed. As `getJson` returns `List[String]`, it loads whole file into memory. We expected a stream of byte chunks/lines here.